paper/paper.md

```diff
@@ -0,0 +1,329 @@
+    orcid: 0000-0002-5294-0198
+    affiliation: 1
+  - name: Kacper Kowalik
+    orcid: 0000-0003-1709-3744
+    affiliation: 1
+  - name: Anna Rosen
+    orcid: 0000-0003-4423-0660
+    affiliation: 2
+affiliations:
+  - name: National Center for Supercomputing Applications, University of Illinois at Urbana-Champaign
+    index: 1
+  - name: Harvard-Smithsonian Center for Astrophysics
+    index: 2
+date: 24 May 2018
+bibliography: paper.bib
+---
+
+# Summary
+
+Software that processes data with physical units or that models real world
+quantities with physical units must have some way of managing units. This might
+be as simple as the convention that all floating point numbers are understood to
+be in the same physical unit system (for example, the SI MKS units
+system). While simple approaches like this do work in practice, they also are
+fraught with possible error, both by programmers modifying the code who
+unintentionally misinterpret the units, and by users of the software who must
+take care to supply data in the correct units or who need to infer the units of
+data returned by the software.
+
+The `unyt` library is designed both to aid quick calculations at an interactive
+python prompt and to be tightly integrated into a larger Python application or
+library. To aid quick calculations, the top-level `unyt` namespace ships with a
+large number of predefined units and physical constants to aid setting up quick
+calculations without needing to look up unit data or the value of a physical
+constant. Using the `unyt` library as an interactive calculation aid only
+requires knowledge of basic Python syntax and awareness of a few of the methods
+of the `unyt_array` class - for example, the `unyt_array.to()` method to convert
+data to a different unit. As the complexity of the usage increases, `unyt`
+provides a number of optional features to aid these cases, including custom unit
+registries containing both predefined physical units as well as user-defined
+units, built-in output to disk via the pickle protocol and to HDF5 files using
+the h5py library [@h5py], and round-trip conversions to create units compatible
+with other popular Python unit libraries.
+
+Physical units in the `unyt` class are defined in terms of the dimensions of the
+unit, a string representation, and a floating point scaling to the MKS unit
+system. Rather than implementing algebra for unit expressions, we rely on the
+`SymPy` symbolic algebra library [@SymPy] to handle symbolic algebraic
+manipulation. The `unyt.Unit` object can represent arbitrary units formed out of
+the seven base dimensions in the SI unit system: time, length, mass,
+temperature, luminance, electric current, and amount of a substance. In
+addition, `unyt` supports forming quantities defined in other unit systems - in
+particular CGS Gaussian units common in astrophysics as well as geometrized
+"natural" units common in relativistic calculations. In addition, `unyt` ships
```

Comment:
I think it might be worth mentioning here cosmology units, too -- unless I missed it, I don't think it's come up anywhere that we can have multiple "code" or "/h" units in unit registries.

Reply:
That actually isn't part of `unyt` right now.

Reply:
To elaborate: the cosmological units in `yt.units` are added dynamically when we load a cosmological dataset. They're not in the default unit registry. That means technically they're a yt feature, not a `yt.units` or `unyt` feature. In principle they could be added to `unyt` itself (they're definitely useful) but we need some API for telling `unyt` what the redshift and what `h` is.

Reply:
Well, perhaps what I should say is that even if `unyt` does not have that as default behavior, it did inform the design decision and the convergent evolution?

Reply:
Ah, sure, I can definitely mention that in this discussion.